infra/src/test/kotlin/fi/fta/geoviite/infra/ui/pagemodel/map/ToolPanel.kt

```diff
@@ -0,0 +1,94 @@
+package fi.fta.geoviite.infra.ui.pagemodel.map
+
+import fi.fta.geoviite.infra.ui.pagemodel.common.E2EInfoBox
+import fi.fta.geoviite.infra.ui.pagemodel.common.E2EViewFragment
+import fi.fta.geoviite.infra.ui.util.ElementFetch
+import fi.fta.geoviite.infra.ui.util.byQaId
+import org.openqa.selenium.By
+
+class E2EToolPanel(elementFetch: ElementFetch) : E2EViewFragment(elementFetch) {
+
+    fun selectToolPanelTab(tabName: String) {
+        clickChild(By.xpath("//div[@qa-id='tool-panel-tabs']/button[span[text() = '$tabName']]"))
+    }
+
+    val layoutKmPostGeneral: E2ELayoutKmPostGeneralInfoBox by lazy {
+        infoBox("km-post-infobox", ::E2ELayoutKmPostGeneralInfoBox)
+    }
+
+    val layoutKmPostLocation: E2ELayoutKmPostLocationInfoBox by lazy {
+        infoBox("layout-km-post-location-infobox", ::E2ELayoutKmPostLocationInfoBox)
+    }
+
+    val trackNumberGeneralInfo: E2ETrackNumberGeneralInfoBox by lazy {
+        infoBox("track-number-infobox", ::E2ETrackNumberGeneralInfoBox)
+    }
+
+    val referenceLineLocation: E2EReferenceLineLocationInfoBox by lazy {
+        infoBox("reference-line-location-infobox", ::E2EReferenceLineLocationInfoBox)
+    }
+
+    val locationTrackGeneralInfo: E2ELocationTrackGeneralInfoBox by lazy {
+        infoBox("location-track-infobox", ::E2ELocationTrackGeneralInfoBox)
+    }
+
+    val locationTrackLocation: E2ELocationTrackLocationInfobox by lazy {
+        infoBox("location-track-location-infobox", ::E2ELocationTrackLocationInfobox)
+    }
+
+    val locationTrackLog: E2ELocationTrackLogInfoBox by lazy {
+        infoBox("location-track-log-infobox", ::E2ELocationTrackLogInfoBox)
+    }
+
+    //re-used old qa-id so don't worry
```

Comment:
I wasn't worried until I read the comment

Reply:
Tää oli jäänne jostain, niin emmäkään uskaltanut koskea 😅